model/src/main/java/uk/gov/hmcts/fees2/register/data/service/impl/FeeServiceImpl.java

```diff
@@ -299,9 +297,16 @@ private Predicate buildFirstLevelPredicate(Root<Fee> fee, CriteriaBuilder builde
             );
         }
 
+        predicates.add(getKeywordPredicate(fee, builder, dto));
+
         return builder.and(predicates.toArray(REF));
 
     }
 
-
+    private Predicate getKeywordPredicate(Root<Fee> fee, CriteriaBuilder builder, LookupFeeDto dto) {
+        if (dto.getKeyword() != null) {
```

Comment:
when keyword is null or empty string then that should be added to criteria like that always. Otherwise it will encounter duplicates when one have keyword and other don't have one

Reply:
That’s what next line is doing and we’ve a test to verify that.